### PR TITLE
LOC 175 Refactor rental and marketplace contracts

### PR DESCRIPTION
**Pull request by maptuhec**

### CODE REVIEW

File: contracts/Property/Rental/Rental.sol [hunk 1]
```diff
@@ -111,28 +137,33 @@ contract Rental is IRental, OwnableUpgradeableImplementation {
...
+        uint[] _refundPercentages,
+        uint[] _daysBeforeStartForRefund,
         bool _isInstantBooking,
-        address _newHostAddress
-    ) public returns(bool success)
+        address _newHostAddress,
+        uint _deposit,
+        uint _minNightsStay,
+        string _rentalTitle
+    ) public onlyValidArraysForCancelation(_refundPercentages, _daysBeforeStartForRefund) returns(bool success)
```
Comment (thcrnk) [inline contracts/Property/Rental/Rental.sol]: Is it a good idea to move this onlyValidArraysForCancelation into the validateUpdate function

File: contracts/Property/Rental/Rental.sol [hunk 2]
```diff
@@ -111,28 +137,33 @@ contract Rental is IRental, OwnableUpgradeableImplementation {
...
     {
         validateUpdate(_rentalId, _newHostAddress);
+        validateRefundPercentages(_refundPercentages);
```
Comment (thcrnk) [inline contracts/Property/Rental/Rental.sol]: Is it a good idea to move this validateRefundPercentages into the validateUpdate function

File: contracts/Property/Rental/RentalFactory/RentalFactory.sol [hunk 3]
```diff
@@ -33,6 +34,14 @@ contract RentalFactory is IRentalFactory, PropertyFactory {
         return rentals[_rentalId];
     }
 
+    function setMarkeplaceId(bytes32 _marketplaceId) public {
```
Comment (thcrnk) [inline contracts/Property/Rental/RentalFactory/RentalFactory.sol]: Put a comment explaining why we need this function as it is kind of hack

File: contracts/Property/Rental/RentalFactory/RentalFactory.sol [hunk 4]
```diff
@@ -48,39 +57,43 @@ contract RentalFactory is IRentalFactory, PropertyFactory {
...
+        string rentalTitle
+		) public returns(bool)
 	{
         require(_hostAddress != address(0));
-        validateCreate(_rentalId, _marketplaceId);
+        validateCreate(_rentalId, getMarketplaceId());
 
-        RentalProxy proxy = new RentalProxy(this);
-        IRental rentalContract = IRental(proxy);
+        // RentalProxy proxy = new RentalProxy(this);
```
Comment (thcrnk) [inline contracts/Property/Rental/RentalFactory/RentalFactory.sol]: Uncomment this and use it after, as solidity is optimizing the code and there is no difference in the gas cost if you do it on single line or not

Comment (maptuhec) [reply contracts/Property/Rental/RentalFactory/RentalFactory.sol]: But it counts it as one more variable and in this way we hit the limit of the variables.

File: contracts/Property/Rental/RentalFactory/RentalFactory.sol [hunk 5]
```diff
@@ -42,6 +43,15 @@ contract RentalFactory is IRentalFactory, PropertyFactory {
         return marketplaceId;
     }
 
+    function getRentalsArrayLenght() public view returns(uint _rentalsArrayLength) {
```
Comment (thcrnk) [inline contracts/Property/Rental/RentalFactory/RentalFactory.sol]: Typo in getRentalsArrayLenght - length

File: contracts/Property/Rental/RentalFactory/RentalFactory.sol [hunk 6]
```diff
@@ -42,6 +43,15 @@ contract RentalFactory is IRentalFactory, PropertyFactory {
...
+       return rentalIds.length;
+    }
+
+    function addRentalToMapping(address _rentalAddress, bytes32 _rentalId) {
```
Comment (thcrnk) [inline contracts/Property/Rental/RentalFactory/RentalFactory.sol]: this should have modifier or has to be internal

Comment (thcrnk) [reply contracts/Property/Rental/RentalFactory/RentalFactory.sol]: Also second check all setters and non-view functions if they has modifiers and who could call them.

**Pull request merged**